lib/timescaledb/rails/extensions/active_record/postgresql_database_tasks.rb

```diff
@@ -145,7 +145,18 @@ def timescale_structure_dump_default_flags
 
         # @return [Boolean]
         def timescale_enabled?
-          Timescaledb::Rails::Hypertable.table_exists?
+          pool_name(connection.pool) == pool_name(Timescaledb::Rails::Hypertable.connection.pool) &&
+            Timescaledb::Rails::Hypertable.table_exists?
+        end
+
+        def pool_name(pool)
+          if pool.respond_to?(:db_config)
+            pool.db_config.name
+          elsif pool.respond_to?(:spec)
+            pool.spec.name
+          else
+            raise "Don't know how to get pool name from #{pool.inspect}"
+          end
```

Comment:
It would be great to find a way to extract this code and avoid the duplication from here and https://github.com/crunchloop/timescaledb-rails/pull/58/files#diff-fcf7b67734152346dd66ba227b705d281ef8010edfe237b18d3d24d47a1801eaL152-R165